crates/turborepo-lib/src/shim.rs

```diff
@@ -475,9 +479,20 @@ impl RepoState {
 
                 // FIXME: We should save this package manager that we detected
                 let workspace_globs =
-                    PackageManager::get_package_manager(path, package_json.as_ref())
+                    match PackageManager::get_package_manager(path, Some(package_json))
                         .and_then(|mgr| mgr.get_workspace_globs(path))
-                        .ok();
+                    {
+                        Ok(globs) => Some(globs),
+                        Err(Error::MultiplePackageManagers { managers }) => {
+                            warn!(
```

Comment:
This is actually the wrong place to print out the warning. This is generating _potential_ `turbo` roots, but we don't actually know if this is going to be the one selected. This could be _entirely_ below the pivot for the eventually-detected root, making the warning message wrong.

So, for this to work, we would need to turn `workspace_globs` into `detect_result` and effectively forward the `Result<PackageManager, Error>`.

Then, at consumption time in `process_potential_turbo_roots`, we know that we got an ambiguous detection. If `process_potential_turbo_roots` is materially impacted by the workspace glob being ambiguous, _then_ we print a warning.